workers/unity/Packages/io.improbable.gdk.testutils/Editor/SpatialDeploymentManager.cs

```diff
@@ -17,21 +17,68 @@ public class SpatialDeploymentManager : IDisposable
     {
         private Process spatialProcess;
 
-        public static Task<SpatialDeploymentManager> Start(string deploymentJsonPath, string snapshotPath)
+        public static async Task<SpatialDeploymentManager> Start(string deploymentJsonPath, string snapshotPath)
         {
             if (!File.Exists(Path.Combine(Common.SpatialProjectRootDir, deploymentJsonPath)))
             {
-                return Task.FromException<SpatialDeploymentManager>(
-                    new ArgumentException($"Could not find deployment config at {deploymentJsonPath}"));
+                throw new ArgumentException($"Could not find deployment config at {deploymentJsonPath}");
             }
 
             if (!File.Exists(Path.Combine(Common.SpatialProjectRootDir, snapshotPath)))
             {
-                return Task.FromException<SpatialDeploymentManager>(
-                    new ArgumentException($"Could not find snapshot at {snapshotPath}"));
+                throw new ArgumentException($"Could not find snapshot at {snapshotPath}");
             }
 
-            var tcs = new TaskCompletionSource<SpatialDeploymentManager>();
+            await BuildWorkerConfigs().ConfigureAwait(false);
+            var process = await StartSpatial(deploymentJsonPath, snapshotPath).ConfigureAwait(false);
+
+            return new SpatialDeploymentManager
+            {
+                spatialProcess = process
+            };
+        }
+
+        private static Task BuildWorkerConfigs()
+        {
+            var tcs = new TaskCompletionSource<bool>();
+
+            var processInfo =
+                new ProcessStartInfo("spatial", "build build-config")
+                {
+                    CreateNoWindow = true,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                    UseShellExecute = false,
+                    WorkingDirectory = Common.SpatialProjectRootDir
+                };
+
+            var process = new Process
```

Comment:
How do we Dispose of the process after use?

Reply:
Could call `Dispose` at the end of the `process.Exited` closure? Although unsure if that's actually a valid thing to do. Will do a bit of digging